### PR TITLE
[AIRFLOW-4876] Making tests in CoreTest rerunnable

### PR DESCRIPTION
**Pull request by exploy**
- all related DagRun, TaskInstance and TaskFail objects are properly removed on tear down
- replaced "models.<ModelName>" importing with "<ModelName>" direct usage
- extracted test specific DAG ids into test suit constants

### Jira

- [x] My PR addresses the following [Airflow Jira](https://issues.apache.org/jira/browse/AIRFLOW/) issues and references them in the PR title. For example, "\[AIRFLOW-XXX\] My Airflow PR"
  - https://issues.apache.org/jira/browse/AIRFLOW-4876
  - In case you are fixing a typo in the documentation you can prepend your commit with \[AIRFLOW-XXX\], code changes always need a Jira issue.
  - In case you are proposing a fundamental code change, you need to create an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)).
  - In case you are adding a dependency, check if the license complies with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).

### Description

Contribution to any project is easier when you can quickly iterate on tests. 

We have spotted that tests in CoreTest are not cleaning properly their state in the DB while performing tear down. 

Steps to reproduce: run test in CoreTest twice and some will fail 

### Commits

- [x] My commits all reference Jira issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
  1. Subject is separated from body by a blank line
  1. Subject is limited to 50 characters (not including Jira issue reference)
  1. Subject does not end with a period
  1. Subject uses the imperative mood ("add", not "adding")
  1. Body wraps at 72 characters
  1. Body explains "what" and "why", not "how"

### Documentation

- [x] In case of new functionality, my PR adds documentation that describes how to use it.
  - All the public functions and the classes in the PR contain docstrings that explain what it does
  - If you implement backwards incompatible changes, please leave a note in the [Updating.md](https://github.com/apache/airflow/blob/master/UPDATING.md) so we can assign it to a appropriate release

### Code Quality

- [x] Passes `flake8`

### CODE REVIEW

File: tests/core.py [hunk 1]
```diff
@@ -88,11 +99,23 @@ def execute(*args, **kwargs):
 
 
 class CoreTest(unittest.TestCase):
+    TEST_SCHEDULE_WITH_NO_PREVIOUS_RUNS_DAG_ID = \
+        TEST_DAG_ID + 'test_schedule_dag_no_previous_runs'
```
Comment (ashb) [inline tests/core.py]: These can fit on a single line - our enforce line length it 110 chars.

**Comment by exploy** [comment]
Happy to address more comments but there is nothing to address right now :)

**Comment by potiuk** [comment]
One comment  - I think it's best to split it to 1-commit per PR (see CONTRIBUTING.md). I know it's a bit of beauracracy overhead, but it allows us to manage distributed environment with multiple contributors more controlled. It would be perfect if you split it to separate PR.

**Comment by exploy** [comment]
Extracted the commit to the separate PR.

**Review by potiuk — COMMENTED** [review]
Looks good to me @ashb maybe you have other comments?

One NIT: I think the linter improvements make TestCore pylint-able - so maybe you can remove it from pylint_todo.txt already ?

**Review by ashb — COMMENTED** [review]
Only comment is that the change from `models.DagBag` to `DagBag` etc make the diff messier than needed. Don't mind it though.

**Comment by exploy** [comment]
> Looks good to me @ashb maybe you have other comments?
> 
> One NIT: I think the linter improvements make TestCore pylint-able - so maybe you can remove it from pylint_todo.txt already ?

The change improves Pylint a bit but does not solve all errors

**Comment by exploy** [comment]
> Only comment is that the change from `models.DagBag` to `DagBag` etc make the diff messier than needed. Don't mind it though.

Wanted to organize the imports so made this change. It may be tricky but I could extract it to separate commit

**Comment by exploy** [comment]
Any more comments? If not it would be great to have it merged

**Review by potiuk — APPROVED** [review]

**Pull request merged**